library/src/main/java/com/github/pwittchen/reactivesensors/library/ReactiveSensors.java

```diff
@@ -134,7 +155,72 @@ public boolean hasSensor(int sensorType) {
         }
       }
     }, strategy).doOnCancel(new Action() {
-      @Override public void run() throws Exception {
+      @Override
+      public void run() throws Exception {
+        sensorManager.unregisterListener(listener);
+      }
+    });
+  }
+
+  /**
+   * Returns RxJava Observable, which allows to monitor hardware sensors as a stream of
+   * ReactiveSensorEvent object with defined sampling period
+   *
+   * @param samplingPeriodInUs sampling period in microseconds,
+   * @param handler the Handler the sensor events will be delivered to, use default if it is null
+   * you can use predefined values from SensorManager class with prefix SENSOR_DELAY
+   * @param strategy BackpressureStrategy for RxJava 2 Flowable type
+   * @param sensorTypes sensor types array from Sensor class from Android SDK
+   * @return RxJava Observable with ReactiveSensorEvent
+   */
+  public Flowable<ReactiveSensorEvent> observeSensors(final int samplingPeriodInUs,
```

Comment:
This method has around ~50 lines of code. It's too many. Please, try to split it into smaller methods, so the max length of the single method will have 20 lines of code or less.